src/Teambuilder/damagecalc.cpp

```diff
@@ -0,0 +1,854 @@
+    ui->mystatus->clear();
+    ui->mygender->clear();
+
+    ui->opoke->clear();
+    ui->oability->clear();
+    ui->oitem->clear();
+    ui->onature->clear();
+    ui->ostatus->clear();
+    ui->ogender->clear();
+
+    QStringList pokeNames = PokemonInfo::Names(pgen, true);
+
+    ui->mypoke->addItems(pokeNames);
+    ui->opoke->addItems(pokeNames);
+
+    QStringList abilityNames = AbilityInfo::Names(pgen);
+
+    ui->myability->addItems(abilityNames);
+    ui->oability->addItems(abilityNames);
+
+    ui->myitem->addItems(ItemInfo::SortedNames(pgen));
+    ui->oitem->addItems(ItemInfo::SortedNames(pgen));
+
+    ui->mynature->addItems(NatureInfo::Names(pgen));
+    ui->onature->addItems(NatureInfo::Names(pgen));
+
+    ui->mystatus->addItems(StatInfo::StatusNames());
+    ui->ostatus->addItems(StatInfo::StatusNames());
+
+    int max = (gen >= 3 ? 31 : 15);
+
+    ui->myhpiv->setMaximum(max);
+    ui->myatkiv->setMaximum(max);
+    ui->mydefiv->setMaximum(max);
+    ui->myspatkiv->setMaximum(max);
+    ui->myspdefiv->setMaximum(max);
+    ui->myspdiv->setMaximum(max);
+
+    ui->ohpiv->setMaximum(max);
+    ui->oatkiv->setMaximum(max);
+    ui->odefiv->setMaximum(max);
+    ui->ospatkiv->setMaximum(max);
+    ui->ospdefiv->setMaximum(max);
+    ui->ospdiv->setMaximum(max);
+
+    ui->myhpiv->setValue(max);
+    ui->myatkiv->setValue(max);
+    ui->mydefiv->setValue(max);
+    ui->myspatkiv->setValue(max);
+    ui->myspdefiv->setValue(max);
+    ui->myspdiv->setValue(max);
+
+    ui->ohpiv->setValue(max);
+    ui->oatkiv->setValue(max);
+    ui->odefiv->setValue(max);
+    ui->ospatkiv->setValue(max);
+    ui->ospdefiv->setValue(max);
+    ui->ospdiv->setValue(max);
+
+    ui->move->setCurrentText(move);
+
+    ui->mypoke->setCurrentText(mypoke);
+    ui->myability->setCurrentText(myability);
+    ui->myitem->setCurrentText(myitem);
+    ui->mynature->setCurrentText(mynature);
+    ui->mystatus->setCurrentText(mystatus);
+    ui->mygender->setCurrentText(mygender);
+
+    ui->opoke->setCurrentText(opoke);
+    ui->oability->setCurrentText(oability);
+    ui->oitem->setCurrentText(oitem);
+    ui->onature->setCurrentText(onature);
+    ui->ostatus->setCurrentText(ostatus);
+    ui->ogender->setCurrentText(ogender);
+
+    ui->weather->setCurrentText(weather);
+
+    ui->myitemlabel->setVisible(gen > 1);
+    ui->myitem->setVisible(gen > 1);
+    ui->myabilitylabel->setVisible(gen > 2);
+    ui->myability->setVisible(gen > 2);
+    ui->mynaturelabel->setVisible(gen > 2);
+    ui->mynature->setVisible(gen > 2);
+    ui->mygenderlabel->setVisible(gen > 2);
+    ui->mygender->setVisible(gen > 2);
+
+    ui->oitemlabel->setVisible(gen > 1);
+    ui->oitem->setVisible(gen > 1);
+    ui->oabilitylabel->setVisible(gen > 2);
+    ui->oability->setVisible(gen > 2);
+    ui->onaturelabel->setVisible(gen > 2);
+    ui->onature->setVisible(gen > 2);
+    ui->ogenderlabel->setVisible(gen > 2);
+    ui->ogender->setVisible(gen > 2);
+
```

Comment:
`ui->label_20->setVisible(gen > 1);
    ui->weather->setVisible(gen > 1);`
is missing. (And I think label_20 should be changed to a more descriptive name.